src/battle_ai_main.c

```diff
@@ -1441,8 +1441,9 @@ static s32 AI_CheckBadMove(u32 battlerAtk, u32 battlerDef, u32 move, s32 score)
         case EFFECT_PRESENT:
         case EFFECT_FIXED_DAMAGE_ARG:
         case EFFECT_FOCUS_PUNCH:
-            // AI_CBM_HighRiskForDamage
-            if (aiData->abilities[battlerDef] == ABILITY_WONDER_GUARD && effectiveness < AI_EFFECTIVENESS_x2)
+            if (HasDamagingMove(battlerDef) && !((gBattleMons[battlerAtk].status2 & STATUS2_SUBSTITUTE)
+            || IsBattlerIncapacitated(battlerDef, aiData->abilities[battlerDef])
+            || gBattleMons[battlerDef].status2 & (STATUS2_INFATUATION | STATUS2_CONFUSION)))
                 ADJUST_SCORE(-10);
             break;
```

Comment:
Could you add this part to the `IsDamageMoveUnusable` function?

Reply:
Generally all damage moves that are bad should be handled in the function to skip damage calcs for them

Reply:
also I think the check for focus punch in `AI_CheckViability` should be removed. It is a damage move that gets judged by the damage. Increasing score here doesn't make sense because it could be increased without being a good damage move in the first place.

Reply:
Didn't know `IsDamageMoveUnusable`, that's sweet. Completely agree with other points.